ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Ledger.hs

```diff
@@ -484,27 +486,28 @@ encodeShelleyLedgerState
                        , shelleyLedgerTransition
                        } =
     encodeVersion serialisationFormatVersion2 $ mconcat [
-        CBOR.encodeListLen 2
+        CBOR.encodeListLen 3
```

Comment:
I assume it is a bug to encode the length as 2, when in fact it contains three items.

Reply:
🤦

Reply:
Since you're changing the serialization scheme for Shelley ledgers, we already have to discard the snapshots, right? So this bugfix itself doesn't add any further breakage.

The encoder and decoder were wrong but compatible with each other, so the only problem here was that the CBOR itself was technically ill-formed. For example, general CBOR tooling would have failed to parse it.

cc: @dcoutts

Reply:
> Since you're changing the serialization scheme for Shelley ledgers, we already have to discard the snapshots, right? So this bugfix itself doesn't add any further breakage.

exactly! in fact, we are bundling several breaking changes to the serialization scheme: the incremental reward calculation, the unaggregated reward update, and the cbor-in-cbor serialization to handle the newly annotated ledger state.